### PR TITLE
Add Offset3D

### PR DESCRIPTION
**Pull request by danieltwagner**
I found this useful for my project and thought it would be a good addition to your repo.
This is my first time tinkering with SDFs so it might not be the best approach. Any feedback much appreciated.

### CODE REVIEW

**Comment by deadsy** [comment]
I used to have one of those:

https://github.com/deadsy/sdfx/commit/62b3f3be4030b7e36c6f69609d7644aa940072fc

I think I got rid of it because it wasn't being used.

**Comment by danieltwagner** [comment]
Oh, interesting. Is there a better way to achieve the same thing? I'm looking for a shell-type operation. Working around this seems to involve a lot of manual work.

File: sdf/sdf3.go [hunk 1]
```diff
@@ -1038,7 +1038,7 @@ type OffsetSDF3 struct {
 	bb     Box3    // bounding box
 }
 
-// Elongate3D returns the elongation of an SDF3.
+// Offset3D enlarges/shrinks an SDF uniformly outwards/inwards from all surfaces
```
Comment (deadsy) [inline sdf/sdf3.go:1041]: It's a bit non-intuitive. Consider a rectangle - An outside offset has rounded corners. An inside offset is rectangular.

Comment (danieltwagner) [reply sdf/sdf3.go:1041]: Definitely. Other operations also have non-intuitive behavior, like `Elongate3D` after `Translate3D` doesn't actually elongate the object you're modeling as you might think but spreads that elongation between the shape and the space from origin. Would be great to compile a list of gotchas like the above!

**Pull request merged**